perception/autoware_tensorrt_bevdet/src/bevdet_node.cpp

```diff
@@ -0,0 +1,235 @@
+  sub_fr_img_.subscribe(this, "~/input/topic_img_fr", rclcpp::QoS{1}.get_rmw_qos_profile());
+
+  sub_bl_img_.subscribe(this, "~/input/topic_img_bl", rclcpp::QoS{1}.get_rmw_qos_profile());
+  sub_br_img_.subscribe(this, "~/input/topic_img_br", rclcpp::QoS{1}.get_rmw_qos_profile());
+
+  sync_ = std::make_shared<Sync>(
+    MySyncPolicy(10), sub_fl_img_, sub_f_img_, sub_fr_img_, sub_bl_img_, sub_b_img_, sub_br_img_);
+
+  sync_->registerCallback(std::bind(&TRTBEVDetNode::callback, this, _1, _2, _3, _4, _5, _6));
+}
+
+void TRTBEVDetNode::startCameraInfoSubscription()
+{
+  // cams: ["CAM_FRONT_LEFT", "CAM_FRONT", "CAM_FRONT_RIGHT", "CAM_BACK_LEFT", "CAM_BACK",
+  // "CAM_BACK_RIGHT"]
+  sub_fl_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_fl/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(0, msg); });
+
+  sub_f_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_f/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(1, msg); });
+
+  sub_fr_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_fr/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(2, msg); });
+
+  sub_bl_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_bl/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(3, msg); });
+
+  sub_b_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_b/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(4, msg); });
+
+  sub_br_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_br/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(5, msg); });
+}
+
+void TRTBEVDetNode::callback(
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_f_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fr_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_bl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_b_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_br_img)
+{
+  cv::Mat img_fl, img_f, img_fr, img_bl, img_b, img_br;
+  std::vector<cv::Mat> imgs;
+  img_fl = cv_bridge::toCvShare(msg_fl_img, "bgr8")->image;
+  img_f = cv_bridge::toCvShare(msg_f_img, "bgr8")->image;
+  img_fr = cv_bridge::toCvShare(msg_fr_img, "bgr8")->image;
+  img_bl = cv_bridge::toCvShare(msg_bl_img, "bgr8")->image;
+  img_b = cv_bridge::toCvShare(msg_b_img, "bgr8")->image;
+  img_br = cv_bridge::toCvShare(msg_br_img, "bgr8")->image;
```

Comment:
Are these ensured to be safe for in-place operation?

In imageTransport, there's a line: cv::cvtColor(imgs[i], imgs[i], cv::COLOR_BGR2RGB);
I **think** OpenCV’s cvtColor handles this safely by creating a new cv::Mat internally.
Do you know if this is safe to do? Or could you check whether it's okay to overwrite the input like this?"

Or, we can just create a copy of the images.